sn_protocol/src/error.rs

```diff
@@ -95,8 +95,8 @@ pub enum Error {
     #[error("Insufficient valid spends found: {0:?}")]
     InsufficientValidSpendsFound(DbcAddress),
     /// Node failed to store spend
-    #[error("Failed to store spend: {0:?}")]
-    FailedToStoreSpend(DbcAddress),
+    #[error("Failed to store spend: {0}")]
+    FailedToStoreSpend(String),
```

Comment:
I think in general we should avoid having strings as the context for Error types. 

Any application (implementing the protocol) trying to act upon these errors should be able to use the context information from them. In this case, the context information is the address of the DBC Spend that couldn't be stored (there could be more information attached to it).

It sounds to me here we are trying to avoid having defined more specific error types by having this String? if so, we should add those additional Error types so applications can correctly detect the different scenarios and in some cases auto correct and retry the operation, having a string won't allow them to do such things.

Reply:
Agreed, there is a bad error management and API overall in this area, I was thinking of reworking it in a follow up PR.